wrangler-core/src/main/java/io/cdap/wrangler/lineage/LineageOperations.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.lineage;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.lineage.field.FieldOperation;
+import io.cdap.cdap.etl.api.lineage.field.FieldTransformOperation;
+import io.cdap.wrangler.api.Directive;
+import io.cdap.wrangler.api.lineage.Lineage;
+import io.cdap.wrangler.api.lineage.Mutation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * This class {@link LineageOperations} generates transformation operations used to generate lineage.
+ */
+public final class LineageOperations {
+  private final Schema inputSchema;
+  private final Schema outputSchema;
+  private final List<Directive> directives;
+
+  /**
+   * A constructor for generating the transformation operations for lineage.
+   *
+   * @param inputSchema A input {@link Schema}.
+   * @param outputSchema A output {@link Schema}.
+   * @param directives A {@link List} of directives.
+   */
+  public LineageOperations(Schema inputSchema, Schema outputSchema, List<Directive> directives) {
+    this.inputSchema = inputSchema;
+    this.outputSchema = outputSchema;
+    this.directives = directives;
+  }
+
+  /**
+   * Generates the list of {@link FieldOperation} required for generating lineage.
+   *
+   * @return a {@link List} of {@link FieldOperation}
+   *
+   * @throws Exception thrown if there is any issue generating field operations.
+   */
+  public List<FieldOperation> generate() throws Exception {
+    // After input and output schema are validated, it's time to extract
+    // all the fields from input and output schema.
+    Set<String> input = inputSchema.getFields().stream()
```

Comment:
We can change the constructor to accept Set<String> as the input and output fields instead of Schema, since in `Wrangler` we validate the input and output schema are not null, so it is better to have this logic there. Doing it here will have IDE warning about NPE

Reply:
fixed!